dvc/external_repo.py

```diff
@@ -72,12 +63,6 @@ def external_repo(
 
     try:
         yield repo
-    except NoRemoteError as exc:
-        raise NoRemoteInExternalRepoError(url) from exc
-    except OutputNotFoundError as exc:
-        if exc.repo is repo:
-            raise NoOutputInExternalRepoError(exc.output, repo.root_dir, url) from exc
-        raise
```

Comment:
I guess we could move this to `dvc.api`(the only place we care about providing backward compatibility for) if we really need these, but it seems not worth it.

Reply:
Actually, let me do just that so this is less potentially contraversial. the `PathMissingError` and `FileMissingError` stuff will be controversial anyway and better to be wrapped.